src/script/view_model/content/AppLockViewModel.ts

```diff
@@ -46,28 +47,41 @@ export enum APPLOCK_STATE {
 
 const APP_LOCK_STORAGE = 'app_lock';
 
+const getInactivityTimeout = () => {
+  const teamState = container.resolve(TeamState);
+  const appLock = teamState.teamFeatures()['app-lock'];
+  if (appLock.status === 'enabled') {
+    return appLock.config.inactivity_timeout_secs;
+  }
+  return null;
+};
+
 const getTimeout = (queryName: string, configName: 'APPLOCK_SCHEDULED_TIMEOUT' | 'APPLOCK_UNFOCUS_TIMEOUT') => {
   const queryTimeout = parseInt(UrlUtil.getURLParameter(queryName), 10);
+  const backendTimeout = getInactivityTimeout();
   const configTimeout = Config.getConfig().FEATURE && Config.getConfig().FEATURE[configName];
```

Comment:
I'm not 100% sure but if we get both timeouts from the backend we might be able to drop this in order to reduce complexity.